l10n_us_check_writing_address/models/res_partner.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Ursa Information Systems <http://www.ursainfosystems.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import api, fields, models, _
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.multi
+    def _display_address(self, without_company=False):
+        '''
+        The purpose of this function is to build and return an address formatted accordingly to the
+        standards of the country where it belongs.
+
+        :param address: browse record of the res.partner to format
+        :returns: the address formatted in a display that fit its country habits (or the default ones
+            if not country is specified)
+        :rtype: string
+        '''
+        # get the information that will be injected into the display format
+        # get the address format
+        if self.street2 == False:
```

Comment:
Hmm I think we can simplify this a bit.

Do I assume correctly that a blank line was being added if using the standard `country_id.address_format` when the `street2` was blank? If so, maybe a lil something like:

```
address_format = self.country_id.address_format
if not self.street2:
    address_format = address_format.replace('%(street2)\n', '')
```

Or maybe slightly more robust with a regex, but I think that we'd be fine seeing as this is specific to US.